lib/logidze/meta.rb

```diff
@@ -0,0 +1,76 @@
+# frozen_string_literal: true
+
+module Logidze # :nodoc:
+  # Provide methods to attach meta information
+  module Meta
+    def with_meta(meta, &block)
+      MetaTransaction.wrap_with(meta, &block)
+    end
+
+    def with_responsible(responsible_id, &block)
+      return yield if responsible_id.nil?
+      meta = { Logidze::History::Version::META_RESPONSIBLE => responsible_id }
+      with_meta(meta, &block)
+    end
+
+    class MetaTransaction # :nodoc:
+      def self.wrap_with(meta, &block)
+        new(meta, &block).perform
+      end
+
+      attr_reader :meta, :block
+
+      delegate :connection, to: ActiveRecord::Base
+
+      def initialize(meta, &block)
+        @meta = meta
+        @block = block
+      end
+
+      def perform
+        return if block.nil?
+        return block.call if meta.nil?
+        ActiveRecord::Base.transaction { call_block_in_meta_context }
+      end
+
+      private
+
+      def call_block_in_meta_context
+        meta_stack.push(meta)
+        pg_set_meta_param(current_meta)
+        result = block.call
+        pg_reset_meta_param
+        result
+      rescue StandardError => exception
```

Comment:
Let's use `begin ... ensure` instead

Reply:
I tried it initially to avoid memoizing and passing down the current meta at the beginning of the block, switching back